lib/libze/libze.c

```diff
@@ -2036,6 +2036,54 @@ typedef struct libze_destroy_cbdata {
     libze_destroy_options *options;
 } libze_destroy_cbdata;
 
+
+typedef struct dependents_cbdata {
+    libze_destroy_cbdata *cbd;
+    zfs_handle_t         *cb_target;
+    nvlist_t             *cb_nvl;
+} dependents_cbdata_t;
+
+
+static int
+promote_dependent_datasets(dependents_cbdata_t *cb)
+{
+    nvpair_t *pair;
+    char *name;
+    pair = nvlist_next_nvpair(cb->cb_nvl, NULL);
+    for ( ; pair != NULL; pair = nvlist_next_nvpair(cb->cb_nvl, pair)) {
```

Comment:
We actually only want to promote one of the clones. All the other clones will then become dependent on that promoted dataset. I'd expect picking the oldest clone probably makes the most sense.